weblate/trans/views/files.py

```diff
@@ -86,18 +86,26 @@ def download_component(request, project, component):
 
 def download_project(request, project):
     obj = get_project(request, project)
-    obj.commit_pending("download", None)
+    components = obj.component_set.filter_access(request.user)
+    if not components:
+        raise PermissionDenied()
```

Comment:
I think it's more appropriate to create empty file in this case. This is what happens right now when there is no component in a project.